AuroraEditor/Base/AppPreferences/Model/Updates/UpdateEditorRepository.swift

```diff
@@ -0,0 +1,328 @@
+        } else {
+            UpdateObservedModel().updateState = .error
+        }
+    }
+
+    /**
+     Handles the completion of a download task and manages the downloaded file.
+
+     This method is called when the download task has finished downloading the file to a temporary location.
+     It checks if there is sufficient disk space available, moves the downloaded file to the 
+     desired location, and logs the outcome.
+
+     - Parameters:
+       - session: The URLSession that initiated the download task.
+       - downloadTask: The URLSessionDownloadTask that completed.
+       - location: The local URL where the downloaded file is temporarily stored.
+
+     - Note:
+       This method checks if a suggested filename is provided by the server, moves the downloaded 
+       file to a specified location, updates the `updateState` property of
+       `WorkspaceManager.shared.updateEditorModel` based on the outcome,
+       and logs any errors that occur during the process.
+     **/
+    func urlSession(_ session: URLSession,
+                    downloadTask: URLSessionDownloadTask,
+                    didFinishDownloadingTo location: URL) {
+        let tempDirectory = self.tempDirectory
+        let appName = "AuroaEditor-\(UUID().uuidString)"
+        let savedFile = "\(appName).zip"
+        let tempLocalURL = tempDirectory.appendingPathComponent(savedFile)
+
+        // Check for available disk space
+        guard isDiskSpaceAvailable(forFileOfSize: requiredStorageSize) else {
+            return
+        }
+
+        do {
+            // Move the downloaded file to the temporary location
+            try FileManager.default.moveItem(at: location, to: tempLocalURL)
+
+            // Unzip the downloaded file
+            let destinationURL = tempDirectory.appendingPathComponent(appName)
+            try FileManager.default.unzipItem(at: tempLocalURL, to: destinationURL)
+
+            self.updateFileUrl = destinationURL.path
+
+            // Construct the final file path
+            if let firstContent = try FileManager.default.contentsOfDirectory(atPath: destinationURL.path).first {
+                let finalFilePath = destinationURL.appendingPathComponent(firstContent)
+
+                // Calculate SHA-256 checksum and verify
+                if let checksum = calculateSHA256Checksum(forFileAtPath: finalFilePath.path) {
+                    DispatchQueue.main.async {
+                        if checksum == self.model.updateModelJson?.sha256sum {
+                            self.model.updateState = .updateReady
+                        } else {
+                            // If for whatever reason possible the checksum is invalid
+                            // we will notify the user in the update page they will be allowed
+                            // to install the update but at their own risk.
+                            self.model.updateState = .checksumInvalid
```

Comment:
Sounds risky to let the user install something that is asserted as not valid.
At the end of the day it's AE that will be blamed.
What do you think?

Reply:
The update service validates the checksum with the one on the server

Reply:
@Maartz what do you suggest I do instead?

Reply:
If the user must click a button with a `.destructive` style, it should be good.
Then the user knows it's a dangerous action to perform.